test/scheduler.test.ts

```diff
@@ -130,4 +130,237 @@ describe("Schedule Store", () => {
+    await expect(store.schedules.get(nonExistingScheduleId)).rejects.toThrowError("Not found");
+  });
+
+  test("Schedule Store: Delete schedule that exists", async () => {
+    const scheduleId = "schedule-to-delete";
+
+    // Create the schedule first
+    await store.schedules.create(
+      scheduleId,
+      scheduleId,
+      "Existing Schedule",
+      "* * * * *",
+      {},
+      "promise-1",
+      60000,
+      {},
+      '{"message": "Test"}',
+      {},
+    );
+
+    // Attempt to delete the schedule
+    const isDeleted = await store.schedules.delete(scheduleId);
+    expect(isDeleted).toBeUndefined();
+
+    // Attempt to get the deleted schedule, should throw NOT_FOUND error
+    await expect(store.schedules.get(scheduleId)).rejects.toThrowError("Not found");
+  });
+
+  test("Schedule Store: Delete schedule that does not exist", async () => {
+    const nonExistingScheduleId = "non-existing-schedule-id";
+
+    // Attempt to delete a schedule that does not exist, should throw NOT_FOUND error
+    await expect(store.schedules.delete(nonExistingScheduleId)).rejects.toThrowError("Not found");
+  });
+  test("Schedule Store: Search by id", async () => {
+    const scheduleId = "search-by-id-schedule";
+
+    // Create the schedule
+    await store.schedules.create(
+      scheduleId,
+      scheduleId,
+      "Search by ID Schedule",
+      "* * * * *",
+      {},
+      "promise-1",
+      60000,
+      {},
+      '{"message": "Test"}',
+      {},
+    );
+
+    // Search for the schedule by id
+    let schedules: Schedule[] = [];
+    for await (const searchResults of store.schedules.search(scheduleId, {})) {
+      schedules = schedules.concat(searchResults);
+    }
+    expect(schedules.length).toBe(1);
+    expect(schedules[0].id).toBe(scheduleId);
+  });
+
+  test("Schedule Store: Search by id with wildcard(s)", async () => {
+    const scheduleIdPrefix = "search-by-id-prefix";
+    const wildcardSearch = `${scheduleIdPrefix}*`;
+
+    // Create multiple schedules with the same prefix
+    for (let i = 1; i <= 3; i++) {
+      await store.schedules.create(
+        `${scheduleIdPrefix}-${i}`,
+        `${scheduleIdPrefix}-${i}`,
+        "Search by ID Prefix Schedule",
+        "* * * * *",
+        {},
+        `promise-${i}`,
+        60000,
+        {},
+        '{"message": "Test"}',
+        {},
+      );
+    }
+
+    // Search for schedules by id prefix with wildcard
+    let schedules: Schedule[] = [];
+    for await (const searchResults of store.schedules.search(wildcardSearch, {})) {
+      schedules = schedules.concat(searchResults);
+    }
+    expect(schedules.length).toBe(3);
+  });
+
+  test("Schedule Store: Search by tags", async () => {
+    const scheduleId = "search-by-tags-schedule";
+    const tags = { category: "search testing", priority: "high" };
+
+    // Create the schedule with specific tags
+    await store.schedules.create(
```

Comment:
Can we create 3 promises that should match and a few that don't, similar to above here?